File: api/src/main/java/com/readrops/api/services/nextcloudnews/NextNewsCredentials.java
package com.readrops.api.services.nextcloudnews;

import com.readrops.api.services.Credentials;

public class NextNewsCredentials extends Credentials {

    public NextNewsCredentials(String login, String password, String url) {
        super(login != null && password != null ? okhttp3.Credentials.basic(login, password) : null, url);
    }
}

File: api/src/main/java/com/readrops/api/services/nextcloudnews/NextNewsCredentials.kt
package com.readrops.api.services.nextcloudnews

import com.readrops.api.services.Credentials

class NextNewsCredentials(login: String?, password: String?, url: String):
        Credentials((login != null && password != null).let {
            okhttp3.Credentials.basic(login!!, password!!)
        }, url)